doc/ConfigurationHeader.md

```diff
@@ -0,0 +1,45 @@
+Configuration Header for *cupla*
+=============================
+
+cupla provides configuration header to use the library without CMake, e.g in a [cling](https://github.com/root-project/cling) environment.
+
+
+Available Pre-compiler Definitions 
+=================================
+
+Pre-compiler definitions can be used to control the behavior of cupla/alpaka.
+The definitions must be passed via a compiler flag or be defined before the accelerator header is included.
+
+- `CUPLA_STREAM_ASYNC_ENABLED`: `0` use synchronous streams, `1` use asynchronous streams
+- `CUPLA_HEADER_ONLY`: `1` *cupla* will be used as header-only library (default), otherwise you must compile all `.cpp` files in [`src/`](https://github.com/ComputationalRadiationPhysics/cupla/tree/master/src)
```

Comment:
Looking at the new header files, it's not really required to set this, given that all of them seem to start with
```c++
#ifndef CUPLA_HEADER_ONLY
#   define CUPLA_HEADER_ONLY 1
#endif
```

Reply:
You are right the documentation should say `The definitions can be passed via a compiler ...`
Is this what you mean?

Reply:
I meant that to use cupla in header-only mode, one can simply include one of the new config headers (e.g. `#include "cupla/config/CpuSerial.hpp"`), without the need to set `CUPLA_HEADER_ONLY` on the compiler command line or with a `#define` directive.

Reply:
Yes, I thought that it is optional can be derived from the documention where we define which behavior is the default `... cupla* will be used as header-only library (default) ...`  
Should I  mention it explicitly to avoid confusion?

Reply:
Ah, now I see what you mean - sorry, I missed the "default" entry.

On a first read, yes, it would be less confusing if it was stated near the top that including a config header is enough to enable this default behaviour.

Reply:
I updated the description

Reply:
Thanks